Task/d42 26459 biz app dbb recipes

nmoisan-d42 opened the pull request:
Recipes for BusinessApp Composition

- Device Impacts on a given set of business app(s)
- Business App Impacted on a given set of device(s)
- Impacts of EOL/EOS Operating Systems
- Impacts of EOL/EOS Hardware

nmoisan-d42 committed:
first drafts of 4 recipes

nmoisan-d42 committed:
Update README.md

nmoisan-d42 committed:
Include comments and put a generic value in the where clause statements

nmoisan-d42 committed:
Device Impacts column cleanup

nmoisan-d42 committed:
include ORDER BY statements

sherriday847 commented:
Make sure all 'where' commands have the "enter_your_xxxx_here" in all the recipes. AND include it in the instructions at the top of each recipe.
Reword the Comments to be more friendly - this is the only word I can think of - this is a cookbook so a little less formal. A little more description as to how/why they would want to use the recipe.
Add a "Cookbook" folder to the Repo and appropriate folders underneath.  Look at the Cookbook link in D42 for folder examples.
Need to add the version requirements at the top too - on the website we have this as a sidebar.
There may later be an "Inspired by...." line if we've taken the recipe from a customer/user.
Look at the website and see if there is any other info that needs to be included in the readme or at the top in the comments like NOTES etc..

nmoisan-d42 committed:
minor format update

nmoisan-d42 committed:
README files

nmoisan-d42 committed:
correct typo

nmoisan-d42 committed:
friendlier descriptions

jtaylor-d42 commented on DBB Cookbook/Business App Composition Recipes/BusinessApp_Dependencies.sql:
```diff
@@ -0,0 +1,24 @@
+/*
+ * Business App Dependencies
+ * This is a simple query to quickly see what infrastructure impacts 1 or more Business App.
```
Minor, should be `impacts one or more Business App`. Numerical values are typically only used for larger double digit values.

jtaylor-d42 commented on DBB Cookbook/Business App Composition Recipes/BusinessApp_Device_Impacts.sql:
```diff
@@ -0,0 +1,37 @@
+/*
+ * Business App Device Impacts
+ * Use this query if you are looking to discover all business apps that are impacted by a specific set of infrastructure devices and the relationships between them.
+ * You can use object_name to look at the source of the impact, or impacted_object to look at the device directly supporting the business application.
```
Remove comma, second clause is not a complete sentence.

nmoisan-d42 replied:
okay. I'm not sure I like `source of the impact` on a second read. Any ideas on how to improve that wording?

jtaylor-d42 commented on DBB Cookbook/Business App Composition Recipes/BusinessApp_Hardware_EOL_EOS_Impact.sql:
```diff
@@ -0,0 +1,33 @@
+/*
+ * Business App Hardware EOL/EOS Impacts
+ * This query will joins up with the Compute DBB to gather information on all BusinessApp Impacting Devices
```
Grammar, "This query will _join_ up"

jtaylor-d42 commented on DBB Cookbook/Business App Composition Recipes/BusinessApp_Hardware_EOL_EOS_Impact.sql:
```diff
@@ -0,0 +1,33 @@
+ * This query will joins up with the Compute DBB to gather information on all BusinessApp Impacting Devices
+ * that are going to hit hardware EOL/EOS. Use this if you need to analyze any of the changes needed when that EOL/EOS date approaches.
+ * You can easily shift the window or focus on either EOL/EOS by editing the included WHERE filter clause. 
+ * TIP: Additional information can easily be pulled in from the compute dbb as well. Try using `compute.*` to see what information are available.
```
Grammar, "Try using `compute.*` to see what information _is_ available"

jtaylor-d42 replied:
Also, instead of using "compute.*", would it make more sense to point them to the Data Dictionary entry for Compute_DBB?

jtaylor-d42 commented on DBB Cookbook/Business App Composition Recipes/BusinessApp_OperatingSystem_EOL_EOS_Impact.sql:
```diff
@@ -0,0 +1,33 @@
+/*
+ * BusinessApp OS EOL/EOS Impacts
+ * This query will combine BusApp Composition and Compute DBBs to gather information on all BusinessApp Impacting Devices
```
Grammar, "Compute DBB" (remove "s" at end of DBB)

jtaylor-d42 commented on README.md:
```diff
@@ -17,3 +17,6 @@ To import a dashboard, simply follow the steps below.
 
+# DBB & DOQL Cookbook
+
+This repository also contains a collection of recipes that highlight how to utilize the Data Building Blocks (DBBs) and DOQL objects within D42 to answer business questions. These recipes will will serve as a starting point to creating your own dashboards within Insights+.
```
Typo, extra "will". "These recipes will _will_ serve as a starting point

jtaylor-d42 reviewed (commented):
Minor grammar and spelling issues.

Also, there is an empty README.md in the DBB Cookbook/Business App Composition Recipes folder. Is that intentional?

nmoisan-d42 committed:
grammer feedback

rename `DBB cookbook` folder to `CookBook`
several grammer issues in descriptions and readmes.
Point to data dictionary instead
